app/models/vm_or_template.rb

```diff
@@ -102,7 +102,7 @@ class VmOrTemplate < ApplicationRecord
 
   has_many                  :scan_histories,    :dependent => :destroy
   has_many                  :lifecycle_events,  :class_name => "LifecycleEvent"
-  has_many                  :advanced_settings, :as => :resource, :dependent => :destroy
+  has_many                  :advanced_settings, -> { where(:resource_type => "VmOrTemplate") }, :as => :resource, :dependent => :destroy
```

Comment:
Is this necessary?  Doesn't the query for advanced_settings already have where resource_id = id AND resource_type = VmOrTemplate?

```
>> vm.advanced_settings
  AdvancedSetting Load (0.3ms)  SELECT "advanced_settings".* FROM "advanced_settings" WHERE "advanced_settings"."resource_id" = $1 AND "advanced_settings"."resource_type" = $2  [["resource_id", 1], ["resource_type", "VmOrTemplate"]]
```

Reply:
you're right, I thought it's needed by ems's association, but it's not